framework/include/materials/MaterialData.h

```diff
@@ -155,8 +158,13 @@ class MaterialData
   ///@}
 
   /**
-   * Resizes the properties to the specified size
-   * @param size The size of the properties to set
+   * Resizes the number of properties to the specified size (including
+   * stateful old and older properties.
+   *
+   * Warning:
+   * If size is greater than the current size, the new properties are set
+   * to NULL except * for the last element - which is set to a blank material
```

Comment:
Can you explain this a little bit more?  It doesn't make sense to me.  The new properties shouldn't be `NULL`... they aren't even pointers.  And the behavior with the "last element" doesn't make sense to me.